LogicLayer/GameEvents/GameEventService.cs

```diff
@@ -0,0 +1,127 @@
+using GameBoard.LogicLayer.GameEvents.Dtos;
+using GameBoard.DataLayer.Entities;
+using GameBoard.DataLayer.Enums;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore;
+
+namespace GameBoard.LogicLayer.GameEvents
+{
+    internal class GameEventService : IGameEventService
+    {
+        private readonly IGameBoardRepository _repository;
+
+        public GameEventService(IGameBoardRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public async Task CreateGameEventAsync(CreateGameEventDto requestedGameEvent)
+        {
+            var creator = await _repository.ApplicationUsers.SingleAsync(
+                ApplicationUser.UserNameEquals(requestedGameEvent.CreatorUserName));
+
+            var creatorParticipation = new GameEventParticipation
+            {
+                ParticipantId = creator.Id,
+                ParticipationStatus = ParticipationStatus.Creator
+            };
+            var gameEvent = new GameEvent
+            {
+                Name = requestedGameEvent.Name,
+                MeetingTime = requestedGameEvent.MeetingTime,
+                Place = requestedGameEvent.Place,
+                Games = requestedGameEvent.Games
+                    .Select(g => new Game { Name = g, GameStatus = GameStatus.ExistsOnTheList}).ToList(),
+                Participations = new List<GameEventParticipation>()
+            };
+
+            gameEvent.Participations.Add(creatorParticipation);
+            _repository.GameEvents.Add(gameEvent);
+
+            await _repository.SaveChangesAsync();
+        }
+
+        public async Task EditGameEventAsync(EditGameEventDto editedEvent)
+        {
+            var gameEventAndGames = await _repository.GameEvents
+                .Where(ge => ge.Id == editedEvent.Id)
+                .Select(ge => new
+                {
+                    GameEvent = ge,
+                    Games = ge.Games.Where(g => g.GameStatus == GameStatus.ExistsOnTheList)
+                })
+                .SingleAsync();
+
+            var gameEvent = gameEventAndGames.GameEvent;
+            var games = gameEventAndGames.Games;
+
+            gameEvent.Name = editedEvent.Name;
+            gameEvent.MeetingTime = editedEvent.MeetingTime;
+            gameEvent.Place = editedEvent.Place;
+
+            foreach (var game in games)
+            {
+                game.GameStatus = GameStatus.RemovedFromTheList;
+                _repository.Games.Update(game);
+            }
+
+            await _repository.SaveChangesAsync();
+
+            _repository.Games.AddRange(
+                editedEvent.Games.Select(
+                        g => new Game {Name = g, GameEventId = editedEvent.Id, GameStatus = GameStatus.ExistsOnTheList})
+                    .ToList());
+
+            await _repository.SaveChangesAsync();
+        }
+
+        private Task<List<GameEventListItemDto>> GetGameEventsWithSamePartitipationStatus(
```

Comment:
Typo.

Reply:
Also, the same comment about `Same` as in `GameEvent` entity.

Reply:
Fixed both.